cirq/vis/heatmap.py

```diff
@@ -268,23 +272,28 @@ def _plot_on_axis(self, ax: plt.Axes, **collection_options: Any) -> mpl_collecti
         return collection
 
     def plot(
-        self, ax: Optional[plt.Axes] = None, **collection_options: Any
+        self, ax: Optional[plt.Axes] = None, **kwargs: Any
     ) -> Tuple[plt.Axes, mpl_collections.Collection]:
         """Plots the heatmap on the given Axes.
         Args:
             ax: the Axes to plot on. If not given, a new figure is created,
                 plotted on, and shown.
-            collection_options: keyword arguments passed to mcoll.PolyCollection().
+            kwargs: The optional keyword arguments are used to temporarily
+                override the values present in the heatmap config. See
+                __init__ for more details on the allowed arguments.
         Returns:
             A 2-tuple ``(ax, collection)``. ``ax`` is the `plt.Axes` that
             is plotted on. ``collection`` is the collection of paths drawn and filled.
         """
         show_plot = not ax
         if not ax:
             fig, ax = plt.subplots(figsize=(8, 8))
-        collection = self._plot_on_axis(ax, **collection_options)
+        original_config = copy.deepcopy(self._config)
+        self.update_config(**kwargs)
+        collection = self._plot_on_axis(ax)
         if show_plot:
             fig.show()
+        self._config = copy.deepcopy(original_config)
```

Comment:
why do you need deepcopy twice? why do you need deepcopy at all?

Reply:
- I need a deepcopy once because I want to store the original state / dict (before `plot` was called) so that I can undo the changes applied by `update_config` call at the end of the function call to `plot`. 
```python
def plot(...):
  original_state = self._config  # If I don't do a deepcopy, this stores a reference to the self._config dict and hence I won't be able to undo the changes without maintaining additional information. 
  self.update_config(**kwargs) # this modifies the self._config state, which i wish to undo at the end of function call. 
  ........ 
  self._config = original_config # I think I can get rid of the deepcopy here
```

Reply:
Removed the second deepcopy.